src/main.cpp

```diff
@@ -1,6 +1,16 @@
 #include <iostream>
+#include "src/game/game.h"
+#include "src/algorithms/cave_cellular_automata.h"
 
 int main() {
-    std::cout << "Hello, World!" << std::endl;
+    Game game;
+    game.print_board();
+    std::shared_ptr<Algorithm> CCA = std::make_shared<CaveCellularAutomata>(50, 3, 5, 1);
+
+    Generator generator = game.get_generator();
+    generator.push_back_algorithm(CCA);
+    generator.generate(game);
+    game.print_board();
```

Comment:
My suggestion is this:
`...
generator.push(CCA);
game.run();
`
when game.run() method calls generator.generate. There maybe added new thing in the future, would not like to see this logic in main